internal/vault/secrets/ephemeral/gcp_oauth2_access_token.go

```diff
@@ -0,0 +1,321 @@
+	LeaseDuration  types.Int64  `tfsdk:"lease_duration"`
+	LeaseStartTime types.String `tfsdk:"lease_start_time"`
+	LeaseRenewable types.Bool   `tfsdk:"lease_renewable"`
+}
+
+// Schema defines this resource's schema which is the data that is available in
+// the resource's configuration, plan, and state
+func (r *GCPOAuth2AccessTokenEphemeralResource) Schema(_ context.Context, _ ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"backend": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Backend to read credentials from.",
+				Required:            true,
+			},
+			"roleset": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Roleset to generate OAuth2 access token for. Mutually exclusive with `static_account` and `impersonated_account`.",
+				Optional:            true,
+			},
+			"static_account": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Static Account to generate OAuth2 access token for. Mutually exclusive with `roleset` and `impersonated_account`.",
+				Optional:            true,
+			},
+			"impersonated_account": schema.StringAttribute{
+				MarkdownDescription: "GCP Secret Impersonated Account to generate OAuth2 access token for. Mutually exclusive with `roleset` and `static_account`.",
+				Optional:            true,
+			},
+			"token": schema.StringAttribute{
+				MarkdownDescription: "The OAuth2 access token.",
+				Computed:            true,
+				Sensitive:           true,
+			},
+			"token_ttl": schema.Int64Attribute{
+				MarkdownDescription: "The TTL of the token in seconds.",
+				Computed:            true,
+			},
+			consts.FieldLeaseID: schema.StringAttribute{
+				MarkdownDescription: "Lease identifier assigned by vault.",
+				Computed:            true,
+			},
+			consts.FieldLeaseDuration: schema.Int64Attribute{
+				MarkdownDescription: "Lease duration in seconds relative to the time in lease_start_time.",
+				Computed:            true,
+			},
+			"lease_start_time": schema.StringAttribute{
+				MarkdownDescription: "Time at which the lease was read, using the clock of the system where Terraform was running.",
+				Computed:            true,
+			},
+			consts.FieldLeaseRenewable: schema.BoolAttribute{
+				MarkdownDescription: "True if the duration of this lease can be extended through renewal.",
+				Computed:            true,
+			},
+		},
+		MarkdownDescription: "Provides an ephemeral resource to generate GCP OAuth2 access tokens from Vault.",
+	}
+
+	base.MustAddBaseEphemeralSchema(&resp.Schema)
+}
+
+// Metadata sets the full name for this resource
+func (r *GCPOAuth2AccessTokenEphemeralResource) Metadata(ctx context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_gcp_oauth2_access_token"
+}
+
+// isRetryableError checks if the error is retryable (e.g., resource not ready yet)
+func isRetryableError(err error) bool {
```

Comment:
This looks like a generic function, can you check to see if we don't already have something like this in other code?